src/main/java/org/sundry/recipes/model/Ingredient.java

```diff
@@ -21,6 +21,19 @@
   @OneToOne(fetch = FetchType.EAGER)
   private UnitOfMeasure uom;
   
+  public Ingredient(String description, BigDecimal amount, UnitOfMeasure uom) {
+    this.description = description;
+    this.amount = amount;
+    this.uom = uom;
+  }
+  
+  public Ingredient(String description, BigDecimal amount, UnitOfMeasure uom, Recipe recipe) {
```

Comment:
Same comment here. An Ingredient should not reference its parent.

Reply:
Recipe table doesn't have reference to Ingredient. Ingredient has a column for Recipe Id. Its in Uni direction relationship.

Reply:
Same question though. Would you even be using an Ingredient without first looking up a recipe? If not, it does not need to be a Unidirectional relationship. If it does, then let the ORM(Hibernate) layer handle the cyclical setting. (Ie leave it null, hibernate should know to hydrate it when you call to persist.)

Reply:
Fixed this. Should Ingredient and UoM also follow the suit...?

Reply:
Apply the same question. Would you be using it without knowing the recipe?

When you're pulling your data into objects you want to try to avoid cyclical references. It can be hard to do if you're using something like neo, but if you are using a traditional RDMS its quite easy as there is typically a parent -> child relationship mapping. It most cases you want only the parent aware of the relationship, now, of course, this is not always the case, but I don't see anything here that is the exception.

In my current usable of Hibernate, the idea has so far been to only have one repository class. In your case, there would only be a recipe repository. 

Let's take a look at UOM and Category's

Would you ever need to reference UOM outside of the context of a recipe? Would you ever want to look up which recipes use a tablespoon? (Likely not,  that's not normally how you would search) So for UOM you would want the recipe to reference it, but you would not want the UOM to reference the recipe(s)

Would you ever need to reference Category outside of the context of a single recipe? Would you ever want to look up which recipes are Italian? (Likely yes)
So for Categories, you would want to be able to walk from Category to Recipe, so the Category object should have a list of recipes. 

Take it one step further. Would you ever want to reference a Recipe out of the context of a single Category? Would you ever want to look up which category a recipe belongs? (Likely Yes) So, for Recipes, you would want to be able to walk from Recipe to Category.

But wait a minute you say, that creates a cyclical dependency. Yes, it does, it's not possible to completely avoid them, we just want to make sure when we have them they are needed. So, in this case, we need to utilize Hibernates Lazy collection ability as well as:
  1: Ensure that the list of Recipes are excluded from Categories String/Equals/Hashcode
  2: Ensure that Category is excluded from Recipes String/Equals/Hashcode

Reply:
I removed the reference. I will dig thru lazy connection and implement it. I will implement the last two points in the next PR for lombok implementation.